### ISSUE TITLE
Latest published Travis binary errors when run with Rust 1.27.0

### ISSUE THREAD

**Issue by WrinklyNinja** [xd009642/tarpaulin#118]
The error is:

```
/home/travis/.cargo/bin/cargo-tarpaulin: error while loading shared libraries: libproc_macro-5a157e6e97a5a7d7.so: cannot open shared object file: No such file or directory
```

I assume libproc_macro is part of the Rust toolchain and that the hash part of its filename changes between releases.  I don't know if there's anything that can be done to avoid this error, but it would be good if the readme mentioned that published binaries were tied to Rust releases, and perhaps indicate which version a binary needs in its filename (e.g. `cargo-tarpaulin-0.6.2-rust-1.26.1-travis.tar.gz`) to make it more obvious. I've just switched to using `cargo install` and caching the result.

**Comment by xd009642** [comment]
Good catch, I've updated the readme to warn about this issue in the installation section and am currently updating the release script to change the name. I'll close once it's updated and pushed

**Comment by xd009642** [comment]
I ended up updating the travis-install.sh to print `WARNING This method is no longer recommended. Use docker or crates.io to install tarpaulin` to stdout after installing from github releases, and with the readme updated hopefully everyone will migrate away. 

I'm going to close this issue now. Bujt if you see anyone in the wild using travis-install recommend they use crates.io or docker to install tarpaulin :+1:

**Comment by Ortham** [comment]
Revisiting this because I rediscovered the travis binaries that are still being published, wondered why I wasn't already using them, and found that I'd filed this issue...

I checked what shared library the published binary for v0.8.6 is linked against, and it no longer refers to any Rust-specific .so files. I also ran a Travis build against Rust 1.35, 1.36 and 1.37 using the same binary, and all worked, so I don't think this is an issue any more. I don't know what changed, but I guess maybe it was stuff getting stabilised?
